### PR TITLE
Backup unit tests

### PR DESCRIPTION
**Pull request by MiaAltieri**
Unit tests for `lib/charms/mongodb/v0/mongodb_backups.py`

Some tests exist for test scenarios, others that are missing are existing in the integration tests

### CODE REVIEW

File: tests/unit/test_mongodb_backups.py [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+
+        self.assertTrue(isinstance(self.harness.charm.unit.status, BlockedStatus))
+
+    @patch_network_get(private_address="1.1.1.1")
+    @patch("charm.snap.SnapCache")
+    @patch("charm.MongoDBBackups._set_config_options")
+    @patch("charm.MongoDBBackups._resync_config_options")
+    @patch("ops.framework.EventBase.defer")
+    def test_s3_credentials_syncing(self, defer, resync, _set_config_options, snap):
+        """Test charm defers when more time is needed to sync pbm."""
```
Comment (welpaolo) [inline tests/unit/test_mongodb_backups.py:418]: Is the description of this function  and the following description correct ? I believe a more specific description will help to quickly understand the purpose of the test.

**Review by welpaolo — APPROVED** [review]
Great work Mia! I left a minor comment. You can also update the copyright date to `2023` in the files you modified.

File: tests/unit/test_mongodb_backups.py (hunk 1 above)
Comment (MiaAltieri) [reply tests/unit/test_mongodb_backups.py:418]: good catch! Must have been a copy paste error I will update

**Review by marceloneppel — APPROVED** [review]
Amazing work, Mia!

**Pull request merged**